### ISSUE TITLE
The train is getting slower and slower.

### ISSUE THREAD

**Issue by yeonhyochoi** [ultralytics/ultralytics#19300]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

hello, i have some trouble when train yolov8 models.
why the train are getting slower and slower..?
i want to train the model using big batch but, i can`t.

![Image](https://github.com/user-attachments/assets/272f309d-79bd-4939-a60f-4e8f3a775795)

![Image](https://github.com/user-attachments/assets/d04a52de-d94c-416e-a895-249444acb9ac)

![Image](https://github.com/user-attachments/assets/6086df56-ce66-4507-9c1e-8080c10eb3fa)

### Environment

Ultralytics 8.3.53 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4090, 23995MiB)
Setup complete ✅ (36 CPUs, 125.5 GB RAM, 440.1/915.3 GB disk)

OS                  Linux-5.15.0-131-generic-x86_64-with-glibc2.31
Environment         Linux
Python              3.9.19
Install             git
RAM                 125.47 GB
Disk                440.1/915.3 GB
CPU                 Intel Core(TM) i9-10980XE 3.00GHz
CPU count           36
GPU                 NVIDIA GeForce RTX 4090, 23995MiB
GPU count           4
CUDA                12.1

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.4.1>=1.8.0
torch               ✅ 2.4.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.19.1>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.8>=2.0.0

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yeonhyochoi, thank you for your interest in Ultralytics 🚀! We understand you’ve encountered issues with training speed and are here to help. An Ultralytics engineer will review this in detail soon 🔍.

In the meantime, here are a few steps you can take to help us debug the issue better:

1. **Ensure You’re Using the Latest Version**  
Please ensure you’re running the latest version of Ultralytics and its dependencies. You can upgrade using the following command:
   ```bash
   pip install -U ultralytics
   ```
   For more details, refer to the [Installation Guide](https://github.com/ultralytics/ultralytics).

2. **Provide a Minimum Reproducible Example (MRE)**  
To expedite debugging, we request a minimal reproducible example ([MRE](https://docs.ultralytics.com/help/minimum-reproducible-example/)). Please ensure the script/file you shared is concise and focuses only on the exact problem. The `default.yaml` file and training script you provided are detailed; however, simplifying this setup to the crux of the issue would help pinpoint the problem faster.

3. **Training Optimization Tips**  
Using large batch sizes can be limited by system memory (GPU/CPU). Please ensure your settings align with our [Training Best Practices](https://docs.ultralytics.com/guides/model-training-tips/). Also, tweaking parameters like `workers`, `batch`, and disabling certain augmentations (e.g., `mosaic: 0` or `copy_paste: 0.0`) might help improve training times.

4. **Monitor GPU Utilization**  
It may also help to check your system's resource usage during training (CPU, GPU, and disk I/O). Tools like `nvidia-smi` (for NVIDIA GPUs) and system profilers can assist in identifying potential bottlenecks.

---

## Environments
For an alternate setup, you might want to explore running your training in one of these environments. They provide out-of-the-box configurations optimized for ML workloads:
- **Notebooks** with free GPU:  
   <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Preconfigured cloud services**:  
   - [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)  
   - [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**:  
   See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

💡 If you'd like additional real-time assistance or wish to connect with the community, try joining our:
- [Discord Server](https://discord.com/invite/ultralytics) 🎧  
- [Discussion Forum](https://community.ultralytics.com/)  
- [Subreddit](https://reddit.com/r/Ultralytics)

---

Thank you again for raising this issue! 🚀 Let us know if you have updates to share regarding your debugging or experiments.

**Comment by Y-T-G** [comment]
Post only the changed settings. Not the whole config file

**Comment by yeonhyochoi** [comment]
> Post only the changed settings. Not the whole config file

which one should I attach to help you analyze it?
During my training session, I experienced a trouble when train the model was fast for the first few epochs and then slowed down, as shown in the image above.

**Comment by Y-T-G** [comment]
I think you should just restart the training. Maybe something is wrong with DDP processes.

**Comment by yeonhyochoi** [comment]
> I think you should just restart the training. Maybe something is wrong with DDP processes.

i restart and start train but..still same.
it happens somtimes.
..
thank you.

**Comment by Y-T-G** [comment]
What's the output of nvidia-smi when it's slow?
